CMSgov/mint-app
# Easi 2550/add user account collaborator

# EASI-2550

## Changes and Description

- Add functionality to add a user account using CEDAR information when a collaborator is added
- Add hasLoggedIn bool to user account
   - If user hasn't logged in, update the information with OKTA instead at the time of login.
- Decorate the context with the JWT, so we we can use that for reaching OKTA endpoints 


## How to test this change
Note, this change should function for both local and CEDAR. It is worth while to check with both
To check Cedar locally, you need to comment out line 127-129 in routes and connect to the vpn
``` go
	if s.environment.Local() || s.environment.Testing() {
		cedarLDAPClient = local.NewCedarLdapClient(s.logger)
	}
```

1. Create model plan.
2. Add a collaborator to the MODEL PLAN
    a. Verify that a user account was added for this collaborator, and that hasLoggedIn == false
3. Log in as that user. (Or send any request in postman)
    a. Verify that the user account has been updated to hasLoggedIn == true, and any missing information is filled in. 
     


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
